.github/workflows/ci.yml

```diff
@@ -0,0 +1,112 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+
+  merge_group:
```

Comment:
does this mean empty list?

Reply:
More specifically `null`, the key just needs to be present to trigger the job on this event.